src/components/UriViewer/UriDownloadButton.js

```diff
@@ -22,6 +22,41 @@ const getMaxDownloadCostNA = bytes => {
   return Utils.formatUSD(downloadPrice)
 }
 
+export const AzureUriDownloadButton = ({ uri, fileName }) => {
+  const [url, setUrl] = useState()
+  const getUrl = () => {
+    if (isAzureUri(uri)) {
+      setUrl(uri)
+    }
+  }
+  useOnMount(() => {
+    getUrl()
+  })
```

Comment:
This looks like a roundabout way to initialize state. If looks like this could be initialized with something like:
```ts
const [url, setUrl] = useState(isAzureUri(uri) ? uri : undefined)
```
https://react.dev/reference/react/useState#usestate

However, given that `url` doesn't change after it's initialized from `uri`, I don't think there actually needs to be a state variable here. It looks like the component could return based on the value of the `uri` prop.

Reply:
refactored this whole component to make sure I wasnt duplicating as much code, let me know if this looks better!